Added a new config option to set a custom notification interval

Sempiternity87 opened the pull request:
Added a configuration option for the notification interval. Based on the original code, I've set the default value to 5 minutes, but this should allow someone to only get notifications every hour (60 minutes), or day (1440), or any other feasible interval.

It's possible that this may spawn another issue to allow setting the TimeUnit as well.

I don't know how to build it to test myself, so I totally understand if this should be put in a different branch. I'll leave that to your discretion.

shayaantx reviewed (changes requested):
can you rebase your commits into one and bump the version.txt file

Sempiternity87 commented:
Do you want me to bump the version to 5.6.5, or keep it at 5.6.4 after syncing?

shayaantx commented:
> Do you want me to bump the version to 5.6.5, or keep it at 5.6.4 after syncing?

5.6.5

Sempiternity87 committed:
Added a new config option to set a custom notification interval (Issue #35).

Config.java
  Adding config option for Notification Interval.

Scheduler.java
  Added the NOTIFICATION_INTERVAL to initApiNotifications.

ApiRequests.java
  Added a getter for the Notification Interval configuration option. Defaulted to the original 5 minutes that was being used previously.

Readme.md
  Updated to include description of what the option is for, and what the default value is.

Updated version.txt

shayaantx commented on README.md:
```diff
@@ -194,6 +194,7 @@ botdarr:
 | EXISTING_ITEM_PATHS_BLACKLIST | If you want content to NOT appear in searches against your library, you can list blacklisted paths here in comma delimited form, and they will be ignored when building up responses | no |
 | MAX_DOWNLOADS_TO_SHOW | The max number of downloads to show. If you set this to any value less than or equal to 0, no downloads will show | yes | 20 |
 | MAX_RESULTS_TO_SHOW | The max number of results to show per search command. If you set this to any value less than 0, the bot won't startup | yes | 20 | 
+| NOTIFICATION_INTERVAL | The number of minutes between each notification. If you set this to any value less than 0, the bot won't startup | yes | 20 | 
```
technically you are just logging the exception, so the bot will startup, but I don't really care :)

Sempiternity87 replied:
Yeah, I wasn't sure about this. I tried to see where the MAX_DOWNLOADS_TO_SHOW and MAX_RESULTS_TO_SHOW would have thrown an error as well, but I didn't see anything specific. I didn't know if the delay function would have a problem with a negative, but I figured it was at least safer to say it wouldn't work, than to have it not work with no explanation.

Sempiternity87 replied:
Oh, I don't know how we both missed this: the default value specified here says 20, but it should be 5, as that's what it is in the ApiRequests.java.
